kpet/run.py

```diff
@@ -103,41 +95,75 @@ def __init__(self, suite, case):
 
 class Host:
     # pylint: disable=too-few-public-methods, too-many-instance-attributes
-    """A host running test suites"""
+    """A host running tests"""
 
     # pylint: disable=redefined-builtin
-    def __init__(self, name, type, suites, tests):
+    def __init__(self, name, type, suites_and_cases):
         """
         Initialize a host run.
 
         Args:
-            name:       Name of the host.
-            type:       Type of the host.
-            suites:     List of suite runs to execute.
-                        Essentially similar to "tests", but uses different
-                        structure.
-            tests:      List of test runs to execute.
-                        Essentially similar to "suites", but uses different
-                        structure.
+            name:               Name of the host.
+            type:               Type of the host.
+            suites_and_cases:   A list of suite-and-cases tuples.
+                                The first element of each tuple being a
+                                kpet.data.Suite instance and the second
+                                element - a list of kpet.data.Case instances.
+                                Represents the suites and their cases to be
+                                executed on the host.
         """
         assert isinstance(type, data.HostType)
-        assert isinstance(suites, list)
-        assert all(isinstance(suite, Suite) for suite in suites)
-        assert isinstance(tests, list)
-        assert all(isinstance(test, Test) for test in tests)
+        assert isinstance(suites_and_cases, list)
+        assert all((isinstance(suite_and_cases, tuple) and
+                    isinstance(suite_and_cases[0], data.Suite) and
+                    isinstance(suite_and_cases[1], list) and
+                    suite_and_cases[1] and
+                    all(isinstance(case, data.Case)
+                        for case in suite_and_cases[1]))
+                   for suite_and_cases in suites_and_cases)
```

Comment:
Maybe the iteration variable should be `suite_and_case`, instead of `suite_and_cases`?

Reply:
It's awkward, but it's correct. Each item in the "suites_and_cases" list is a tuple with one suite and multiple cases. The naming scheme trying to follow natural English and the absence of a dedicated type for this data produce such names.